app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -542,8 +543,17 @@ def create_vm(args = { })
       raise e
     end
 
+    def unassigned_volumes?(vols)
+      vols&.map do |vol|
+        return true unless vol.key?(:controller_key)
+      end
+      false
+    end
+
     def new_vm(args = {})
       args = parse_args args
+      args = args.deep_symbolize_keys
+      args[:scsi_controllers] = [] if !args.key?(:scsi_controllers) && !args[:volumes].empty? && !unassigned_volumes?(args[:volumes])
```

Comment:
A comment explaining why do you need to set `scsi_controllers` to an empty array would be nice here.